lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2672,3 +2845,42 @@ func (d *zfs) RenameVolumeSnapshot(vol Volume, newSnapshotName string, op *opera
 
 	return nil
 }
+
+// FillVolumeConfig populate volume with default config.
+func (d *zfs) FillVolumeConfig(vol Volume) error {
+	// Copy volume.* configuration options from pool.
+	err := d.fillVolumeConfig(&vol)
```

Comment:
I think we need something like the LVM driver does here:

```go
	// Copy volume.* configuration options from pool.
	// Exclude "block.filesystem" and "block.mount_options" as they depend on volume type (handled below).
	// Exclude "lvm.stripes", "lvm.stripes.size" as they only work on non-thin storage pools (handled below).
	err := d.fillVolumeConfig(&vol, "block.filesystem", "block.mount_options", "lvm.stripes", "lvm.stripes.size")
	if err != nil {
		return err
	}
```

Reply:
Otherwise the block volume settings will get copied from the pool even when not relevant (i.e VM volumes).